```pytb
________________ test_bad_indexed_sensor_state_via_mqtt[pyloop] ________________
[gw14] linux -- Python 3.8.9 /nix/store/q6gfck5czr67090pwm53xrdyhpg6bx67-python3-3.8.9/bin/python3.8

hass = <homeassistant.core.HomeAssistant object at 0x7ffdd41fad60>
mqtt_mock = <MagicMock id='140728066697392'>, setup_tasmota = None

    async def test_bad_indexed_sensor_state_via_mqtt(hass, mqtt_mock, setup_tasmota):
        """Test state update via MQTT where sensor is not matching configuration."""
        config = copy.deepcopy(DEFAULT_CONFIG)
        sensor_config = copy.deepcopy(BAD_INDEXED_SENSOR_CONFIG_3)
        mac = config["mac"]

        async_fire_mqtt_message(
            hass,
            f"{DEFAULT_PREFIX}/{mac}/config",
            json.dumps(config),
        )
        await hass.async_block_till_done()
        async_fire_mqtt_message(
            hass,
            f"{DEFAULT_PREFIX}/{mac}/sensors",
            json.dumps(sensor_config),
        )
        await hass.async_block_till_done()

        state = hass.states.get("sensor.tasmota_energy_apparentpower_0")
        assert state.state == "unavailable"
        assert not state.attributes.get(ATTR_ASSUMED_STATE)
        state = hass.states.get("sensor.tasmota_energy_apparentpower_1")
        assert state.state == "unavailable"
        assert not state.attributes.get(ATTR_ASSUMED_STATE)
        state = hass.states.get("sensor.tasmota_energy_apparentpower_2")
        assert state.state == "unavailable"
        assert not state.attributes.get(ATTR_ASSUMED_STATE)

        async_fire_mqtt_message(hass, "tasmota_49A3BC/tele/LWT", "Online")
        state = hass.states.get("sensor.tasmota_energy_apparentpower_0")
        assert state.state == STATE_UNKNOWN
        assert not state.attributes.get(ATTR_ASSUMED_STATE)
        state = hass.states.get("sensor.tasmota_energy_apparentpower_1")
        assert state.state == STATE_UNKNOWN
        assert not state.attributes.get(ATTR_ASSUMED_STATE)
        state = hass.states.get("sensor.tasmota_energy_apparentpower_2")
        assert state.state == STATE_UNKNOWN
        assert not state.attributes.get(ATTR_ASSUMED_STATE)

        # Test periodic state update
        async_fire_mqtt_message(
            hass, "tasmota_49A3BC/tele/SENSOR", '{"ENERGY":{"ApparentPower":[1.2,3.4,5.6]}}'
        )
        state = hass.states.get("sensor.tasmota_energy_apparentpower_0")
        assert state.state == "1.2"
        state = hass.states.get("sensor.tasmota_energy_apparentpower_1")
        assert state.state == "3.4"
        state = hass.states.get("sensor.tasmota_energy_apparentpower_2")
        assert state.state == "5.6"

        # Test periodic state update with too few values
>       async_fire_mqtt_message(
            hass, "tasmota_49A3BC/tele/SENSOR", '{"ENERGY":{"ApparentPower":[7.8,9.0]}}'
        )

tests/components/tasmota/test_sensor.py:288:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests/common.py:357: in async_fire_mqtt_message
    hass.data["mqtt"]._mqtt_handle_message(msg)
/nix/store/q6gfck5czr67090pwm53xrdyhpg6bx67-python3-3.8.9/lib/python3.8/unittest/mock.py:1081: in __call__
    return self._mock_call(*args, **kwargs)
/nix/store/q6gfck5czr67090pwm53xrdyhpg6bx67-python3-3.8.9/lib/python3.8/unittest/mock.py:1085: in _mock_call
    return self._execute_mock_call(*args, **kwargs)
/nix/store/q6gfck5czr67090pwm53xrdyhpg6bx67-python3-3.8.9/lib/python3.8/unittest/mock.py:1155: in _execute_mock_call
    return self._mock_wraps(*args, **kwargs)
homeassistant/components/mqtt/__init__.py:897: in _mqtt_handle_message
    self.hass.async_run_hass_job(
homeassistant/core.py:424: in async_run_hass_job
    hassjob.target(*args)
homeassistant/util/logging.py:154: in wrapper
    log_exception(format_err, *args)
homeassistant/util/logging.py:152: in wrapper
    func(*args)
/nix/store/5x8d3hsaw5njd3vlb6pl2q0gpr6sl2cx-python3.8-hatasmota-0.2.11/lib/python3.8/site-packages/hatasmota/sensor.py:260: in state_message_received
    state = get_value_by_path(msg.payload, self._cfg.value_path)
/nix/store/5x8d3hsaw5njd3vlb6pl2q0gpr6sl2cx-python3.8-hatasmota-0.2.11/lib/python3.8/site-packages/hatasmota/utils.py:167: in get_value_by_path
    return get_by_path(status, path)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

root = {'ENERGY': {'ApparentPower': [7.8, 9.0]}}
items = ['ENERGY', 'ApparentPower', 2]

    def get_by_path(root, items):
        """Access a nested object in root by item sequence."""
>       return reduce(operator.getitem, items, root)
E       IndexError: list index out of range

/nix/store/5x8d3hsaw5njd3vlb6pl2q0gpr6sl2cx-python3.8-hatasmota-0.2.11/lib/python3.8/site-packages/hatasmota/utils.py:36: IndexError
```

--- a/pkgs/development/python-modules/pysonos/default.nix
+++ b/pkgs/development/python-modules/pysonos/default.nix
@@ -14,7 +14,7 @@
 
 buildPythonPackage rec {
   pname = "pysonos";
-  version = "0.0.44";
+  version = "0.0.45";
 
   disabled = !isPy3k;
 
@@ -23,7 +23,7 @@ buildPythonPackage rec {
     owner = "amelchio";
     repo = pname;
     rev = "v${version}";
-    sha256 = "108818mkb037zs4ikilrskfppcbmqslsm6zaxmy8pphjh7c299mz";
+    sha256 = "0wzmrd9ja5makvsgf0ckil99wr8vw91dml8fi9miiq4la0100q0n";
   };
 
   propagatedBuildInputs = [ ifaddr requests xmltodict ];

--- a/pkgs/servers/home-assistant/component-packages.nix
+++ b/pkgs/servers/home-assistant/component-packages.nix
@@ -2,7 +2,7 @@
 # Do not edit!
 
 {
-  version = "2021.5.1";
+  version = "2021.5.2";
   components = {
     "abode" = ps: with ps; [ abodepy ];
     "accuweather" = ps: with ps; [ accuweather ];
@@ -375,7 +375,7 @@
     "hvv_departures" = ps: with ps; [ ]; # missing inputs: pygti
     "hydrawise" = ps: with ps; [ hydrawiser ];
     "hyperion" = ps: with ps; [ hyperion-py ];
-    "ialarm" = ps: with ps; [ ]; # missing inputs: pyialarm
+    "ialarm" = ps: with ps; [ pyialarm ];
     "iammeter" = ps: with ps; [ ]; # missing inputs: iammeter
     "iaqualink" = ps: with ps; [ iaqualink ];
     "icloud" = ps: with ps; [ pyicloud ];

--- a/pkgs/servers/home-assistant/default.nix
+++ b/pkgs/servers/home-assistant/default.nix
@@ -114,7 +114,7 @@ let
   extraBuildInputs = extraPackages py.pkgs;
 
   # Don't forget to run parse-requirements.py after updating
-  hassVersion = "2021.5.1";
+  hassVersion = "2021.5.2";
 
 in with py.pkgs; buildPythonApplication rec {
   pname = "homeassistant";
@@ -133,7 +133,7 @@ in with py.pkgs; buildPythonApplication rec {
     owner = "home-assistant";
     repo = "core";
     rev = version;
-    sha256 = "0bipjfkz4zqhy84jgrn3qxvs4nxya3j08lcsq3xa31xfz8wnpxwj";
+    sha256 = "1v1ah0kdkhx0pkma2qnk56iv00r2lb4i3sfpwlcr64gfkpzpakv8";
   };
 
   # leave this in, so users don't have to constantly update their downstream patch handling

